### DISCUSSION TITLE
Are data transformations done by r supported or is there a way to use them along with sql models?

### DISCUSSION THREAD

**Discussion by anishjoni** [TobikoData/sqlmesh#3128] [Q&A] [answered]
I am wondering if I can use R transformations with sqlmesh.

My pipeline starts with sqlserver but I do most of my data transformations in r and the feature I want to use sqlmesh for is column level lineage.

If I can export my transformed data tables as csv, can I make a sqlmesh project with sql models of raw data and connect it with a seed model that has the transformed data tables?

Will I be able to see the lineage from sql source tables flowing through the transformed tables created by r(potentially loaded as a seed model)?

**Comment by georgesittas** [comment, marked as answer]
Leaving Toby's [Slack answer](https://tobiko-data.slack.com/archives/C044BRE5W4S/p1726245257925879) here for posterity:

> You can only get column level lineage through sql. We can't compute column level lineage in Python or R because you can do anything in those computations. But if you have csvs and write sql models in SQLMesh, you will get column level lineage. You could probably use R in SQLMesh by using rpy2 as a python model, and returning a pandas dataframe. So create a Python model that uses rpy2 to execute R and return a pandas dataframe. Then use SQL for other transformations.
